spring-boot-autoconfigure/src/test/java/org/axonframework/springboot/autoconfig/InterceptorConfigurationTest.java

```diff
@@ -0,0 +1,512 @@
+        }
+
+        @Bean
+        public MyEventDispatchInterceptor myThirdEventDispatchInterceptor(
+                @Qualifier("eventDispatchInterceptorInvocations") CountDownLatch eventDispatchInterceptorInvocations, 
+                @Qualifier("eventDispatchingInterceptingOutcome") Queue<String> eventDispatchingInterceptingOutcome) {
+            return new MyEventDispatchInterceptor("Unordered", eventDispatchInterceptorInvocations, eventDispatchingInterceptingOutcome);
+        }
+
+        @Bean
+        public CommandHandlingComponent commandHandlingComponent() {
+            return new CommandHandlingComponent();
+        }
+
+        @Bean
+        public QueryHandlingComponent queryHandlingComponent() {
+            return new QueryHandlingComponent();
+        }
+
+        @Bean
+        public EventHandlingComponent eventHandlingComponent() {
+            return new EventHandlingComponent();
+        }
+
+        public static class MyCommandHandlerInterceptor extends MyHandlerInterceptor<CommandMessage<?>> {
+            public MyCommandHandlerInterceptor(String name, CountDownLatch invocation, Queue<String> handlingOutcome) {
+                super(name, invocation, handlingOutcome);
+            }
+        }
+
+        public static class MyQueryHandlerInterceptor extends MyHandlerInterceptor<QueryMessage<?, ?>> {
+            public MyQueryHandlerInterceptor(String name, CountDownLatch invocation, Queue<String> handlingOutcome) {
+                super(name, invocation, handlingOutcome);
+            }
+        }
+
+        public static class MyEventHandlerInterceptor extends MyHandlerInterceptor<EventMessage<?>> {
+            public MyEventHandlerInterceptor(String name, CountDownLatch invocation, Queue<String> handlingOutcome) {
+                super(name, invocation, handlingOutcome);
+            }
+        }
+
+        public static class MyCommandDispatchInterceptor extends MyDispatchInterceptor<CommandMessage<?>> {
+            public MyCommandDispatchInterceptor(String name, CountDownLatch invocation, Queue<String> handlingOutcome) {
+                super(name, invocation, handlingOutcome);
+            }
+        }
+
+        public static class MyQueryDispatchInterceptor extends MyDispatchInterceptor<QueryMessage<?, ?>> {
+            public MyQueryDispatchInterceptor(String name, CountDownLatch invocation, Queue<String> handlingOutcome) {
+                super(name, invocation, handlingOutcome);
+            }
+        }
+
+        public static class MyEventDispatchInterceptor extends MyDispatchInterceptor<EventMessage<?>> {
+            public MyEventDispatchInterceptor(String name, CountDownLatch invocation, Queue<String> handlingOutcome) {
+                super(name, invocation, handlingOutcome);
+            }
+        }
+
+        public static class MyHandlerInterceptor<T extends Message<?>> implements MessageHandlerInterceptor<T> {
```

Comment:
Can we also add a bean of type `MyHandlerInterceptor`? I think we need to make sure that generic `MessageHandlerInterceptors` are registered on all infrastructure components supporting message handler interceptor registration.

Reply:
Thanks for adding those!!